### DISCUSSION TITLE
Namron 4512757 z-wave thermostat not able to set even/whole temperatures

### DISCUSSION THREAD

**Discussion by tronsdal** [zwave-js/zwave-js#6791] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
When I try to set the temp to whole degrees (e.g 17.0, 18.0 etc.) there is an error, and nothing gets updated. If I set it to 17.1, the thermostat gets updated. The same with 17.01 (the thermostat then shows 17.0 in the physical display). I have tried two different thermostats with and without security, and I have tried both with the original z-wave-js and with z-wave-js-ui.

This is very unfortunate, bacause I have a smart integration with my electricity company Tibber, but they only send whole degrees, thus no smart change of temperatures are working. I also have a Homey Pro 2019, and there everything works.

What did you expect to happen instead?
The temp setpoint set to 17.0

Try to set the Namron Thermostat entity to an even number like 17 or 17.0 fails. Half degrees like 17.5 works.

Anything else we should know?

### Software versions

Driver (node-zwave-js): 12.5.5

Z-Wave JS UI: 9.11.1

Home Assistant Z-Wave Integration: Not sure (1.35.0-beta.1 ?)

Home Assistant Z-Wave JS Addon: 3.5.0

ioBroker.zwave2 Adapter: Aeotec Z-Stick Gen 5

If you are using something non-standard, tell us here: nope

### Device information

Manufacturer: Namron
Model name: 16 A Thermostat - White - 4512757
Node ID: 034 (insecure) and 037 (secure)

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-js-ui-store.zip](https://github.com/zwave-js/node-zwave-js/files/15122452/zwave-js-ui-store.zip)

**Comment by AlCalzone** [comment, marked as answer]
I've seen this before with other thermostats. Essentially they expect the setpoint to always have exactly one decimal digit, so 17.0 instead of 17.

**Reply by AlCalzone** [reply]
Escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/6792

**Reply by AlCalzone** [reply]
@tronsdal The fix has been merged, but I'll likely not make a new release until the week after the next since I'll be traveling.
If you want to test it before that, you can follow [these steps](https://zwave-js.github.io/zwave-js-ui/#/device-support/custom-device-files?id=using-custom-device-configuration-files) with [this file](https://raw.githubusercontent.com/zwave-js/node-zwave-js/e802ca03a11736e953666c481ebcca31f80c1e88/packages/config/config/devices/0x0438/4512757.json).

**Reply by tronsdal** [reply]
Great! Thank you so much. That worked perfectly :-)
